src/lib.rs

```diff
@@ -135,13 +142,15 @@ pub fn exponential_backoff(base_wait_time: usize, n: usize, max: usize) -> usize
 }
 
 async fn download_async(
+    py: Python<'_>,
     url: String,
     filename: String,
     max_files: usize,
     chunk_size: usize,
     parallel_failures: usize,
     max_retries: usize,
     input_headers: Option<HashMap<String, String>>,
+    callback: Option<Py<PyAny>>,
```

Comment:
You could type the callback to something more meaningful

Reply:
I didn't find. Tried with pyo3 `PyFunction` but it prevents bounded methods to be passed (so `tqdm().update` doesn't work for instance which was so original goal). Is there a better way to type this as a `Callable` in the context of pyo3 ?

Reply:
Ah true, looking at the docs you're using the correct object. Welp